### PR TITLE
[11.0][MIG] sale_triple_discount

### PR DESCRIPTION
**Pull request by xavierjimenez**
Migration of sale_triple_discount module to version 11.0

Depends on:

- [x] https://github.com/OCA/account-invoicing/pull/370 account_invoicing_triple_discount

### CODE REVIEW

**Comment by pedrobaeza** [comment]
You have added a .DS_store file

**Review by misern2 — CHANGES_REQUESTED** [review]
You should update the security groups on the account_invoice views for the new fields.

Check core code for the discount field: 

https://github.com/odoo/odoo/blob/00ef138ce038d4f8e5a9b7ad2f23ab483e282ebb/addons/sale/views/sale_views.xml#L718-L738

**Comment by rafaelbn** [comment]
Thanks @xavierjimenez ! Please rebase as https://github.com/OCA/account-invoicing/pull/370 has been merge and then we could test this one! :smile:

**Comment by pedrobaeza** [comment]
No need to rebase, as it's in another repository. Just to restart CIs.

**Comment by xavierjimenez** [comment]
I'm not sure how to do that, do I have to open a new PR?

**Comment by pedrobaeza** [comment]
I have restarted both.

**Comment by rafaelbn** [comment]
There are missing dependencies

```
2018-04-18 16:06:05,315 5431 WARNING openerp_test odoo.modules.loading: Transient module states were reset
2018-04-18 16:06:05,315 5431 ERROR openerp_test odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
File "/home/travis/OCB-11.0/odoo/modules/registry.py", line 85, in new
odoo.modules.load_modules(registry._db, force_demo, status, update_module)
File "/home/travis/OCB-11.0/odoo/modules/loading.py", line 308, in load_modules
modules.button_install()
File "<decorator-gen-39>", line 2, in button_install
File "/home/travis/OCB-11.0/odoo/addons/base/module/module.py", line 71, in check_and_log
return method(self, *args, **kwargs)
File "/home/travis/OCB-11.0/odoo/addons/base/module/module.py", line 397, in button_install
modules._state_update('to install', ['uninstalled'])
File "/home/travis/OCB-11.0/odoo/addons/base/module/module.py", line 362, in _state_update
raise UserError(_("You try to install module '%s' that depends on module '%s'.\nBut the latter module is not available in your system.") % (module.name, dep.name,))
odoo.exceptions.UserError: ("You try to install module 'sale_triple_discount' that depends on module 'account_invoice_triple_discount'.\nBut the latter module is not available in your system.", '')
2018-04-18 16:06:05,316 5431 CRITICAL openerp_test odoo.service.server: Failed to initialize database `openerp_test`.
Traceback (most recent call last):
File "/home/travis/OCB-11.0/odoo/service/server.py", line 925, in preload_registries
registry = Registry.new(dbname, update_module=update_module)
File "/home/travis/OCB-11.0/odoo/modules/registry.py", line 85, in new
odoo.modules.load_modules(registry._db, force_demo, status, update_module)
File "/home/travis/OCB-11.0/odoo/modules/loading.py", line 308, in load_modules
modules.button_install()
File "<decorator-gen-39>", line 2, in button_install
File "/home/travis/OCB-11.0/odoo/addons/base/module/module.py", line 71, in check_and_log
return method(self, *args, **kwargs)
File "/home/travis/OCB-11.0/odoo/addons/base/module/module.py", line 397, in button_install
modules._state_update('to install', ['uninstalled'])
File "/home/travis/OCB-11.0/odoo/addons/base/module/module.py", line 362, in _state_update
raise UserError(_("You try to install module '%s' that depends on module '%s'.\nBut the latter module is not available in your system.") % (module.name, dep.name,))
odoo.exceptions.UserError: ("You try to install module 'sale_triple_discount' that depends on module 'account_invoice_triple_discount'.\nBut the latter module is not available in your system.", '')
```

**Comment by murtuzasaleh** [comment]
@xavierjimenez  I am starting working on it.
cc @pedrobaeza

**Comment by xavierjimenez** [comment]
@murtuzasaleh I think the module is working fine, there's a dependency issue with another module (account_invoice_triple_discount) but that module is already merged so I'm not sure how I can solve that.

**Comment by pedrobaeza** [comment]
Is the repository put in oca_dependencies.txt in root folder?

**Comment by pedrobaeza** [comment]
@xavierjimenez can you check what I have said?

**Comment by xavierjimenez** [comment]
I just checked it out and it's not, can I add it?

**Comment by pedrobaeza** [comment]
Yes, of course. That's the idea when you add a new module (migrated or totally new) that depends on modules on other repositories. Doing so, you will get the module available and Travis/runbot will be OK.

**Comment by pedrobaeza** [comment]
All green! @murtuzasaleh  @JayVora-SerpentCS can you please review this one now that the problem has gone?

File: sale_triple_discount/README.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+You can also use negative values to make a charge instead of a discount:
+
+Unit price: 600.00 ->
+
+  - Disc. 1 = 50% -> Amount = 300.00
+  - Disc. 2 = -5% -> Amount = 315.00
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/167/10.0
```
Comment (JayVora-SerpentCS) [inline sale_triple_discount/README.rst]: 11.0 plz

File: sale_triple_discount/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+{
+    'name': 'Sale Triple Discount',
+    'version': '11.0.1.0.0',
+    'category': 'Sales Management',
+    'author': 'QubiQ, '
+              'ADHOC SA, '
+              'Agile Business Group, '
+              'Tecnativa, '
+              'Odoo Community Association (OCA)',
```
Comment (JayVora-SerpentCS) [inline sale_triple_discount/__manifest__.py:13]: Authorname should be added when you added a feature my friend. If not, add your name to contributors in readme.

File: sale_triple_discount/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    'website': 'https://odoo-community.org',
```
Comment (JayVora-SerpentCS) [inline sale_triple_discount/__manifest__.py]: website  -> https://github.com/OCA/sale-workflow

**Review by JayVora-SerpentCS — CHANGES_REQUESTED** [review]

File: sale_triple_discount/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    'license': 'AGPL-3',
+    'summary': 'Manage triple discount on sale order lines',
+    'depends': [
+        'sale',
```
Comment (murtuzasaleh) [inline sale_triple_discount/__manifest__.py]: IMO it should be sale_management instead of a sale module.

**Comment by xavierjimenez** [comment]
Changes done! I'll make the squash with the previous commits when it's ready to merge.

**Comment by rafaelbn** [comment]
Hi @murtuzasaleh @JayVora-SerpentCS @misern2 this PR is all green and attended comment.

@xavierjimenez just take a look in:

- [FIX] https://github.com/OCA/sale-workflow/pull/544 (this could happens also in v11)
- [IMP] https://github.com/OCA/sale-workflow/pull/671

Thanks a lot for your :green_apple: work! :+1: 

cc @chienandalu

**Comment by rafaelbn** [comment]
Hi @xavierjimenez I've tested it and my doubt is about report. Does it have any sense for you?

Take a look:

ex1

---

ex2

**Review by rafaelbn — COMMENTED** [review]
I'm not blocking but IMHO report (PDF of SO) must be improved

**Comment by xavierjimenez** [comment]
Hi @rafaelbn, thanks for the review!

I agree, the report needs a change so I'll try to add the others discounts on the report if they are set.

**Comment by pedrobaeza** [comment]
@xavierjimenez I have just discovered the field `price_reduce`: https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py#L783, that contains the price with the discount. I'm not sure if this field is used everywhere, but maybe changing the computing of it can be enough.

Can you please try?

**Comment by rafaelbn** [comment]
Hello @xavierjimenez @misern2 !

Are you using this module in production enviroment?

Take care for fixes in v10 that apply in this PR

- FIX https://github.com/OCA/sale-workflow/pull/544
- Rework https://github.com/OCA/sale-workflow/pull/671
- FIX https://github.com/OCA/sale-workflow/pull/740

We could try to push this PR reviewing fixes of v10, rebasing this PR and checking again.

Thanks

**Comment by rousseldenis** [comment]
Closing this as very old and no answer from author. Please feel free to reopen it if needed.

**Pull request closed without merging**